### PR TITLE
Add intel-opencl-clang

### PR DESCRIPTION
**Pull request by isuruf**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge/staged-recipes, ready for review.

File: recipes/opencl-clang/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+  host:
+    - llvmdev   {{ llvm_version }}
+    - llvm      {{ llvm_version }}
+    - clangdev  {{ llvm_version }}
+    - llvm-spirv   {{ llvm_version }}*
```
Comment (xhochy) [inline recipes/opencl-clang/meta.yaml]: Is the `*` needed? This is done automatically by conda normally.

Comment (isuruf) [reply recipes/opencl-clang/meta.yaml]: Fixed

**Comment by isuruf** [comment]
Thanks @xhochy for the review.

**Pull request merged**